index.js

```diff
@@ -11,11 +11,21 @@ function toObject(val) {
 	return Object(val);
 }
 
-function isFunc(val) {
-	return typeof val === 'function';
+function assignKey(to, from, key) {
+	var val = from[key];
+
+	if (typeof val === 'undefined' || val === null) {
+		return;
+	}
+
+	if (typeof to[key] === 'undefined' || !isObj(val)) {
+		to[key] = val;
+	} else {
+		to[key] = assign(Object(to[key]), from[key]);
+	}
```

Comment:
@schnittstabil Yes, this is more like a deepAssign implementation.

Just for the record, I made a mistake when I wanted to use deepAssign to deep clone an object so I have to switch to `lodash.cloneDeep` or something like that but that's another story.

But to be honest, the `isArray` part confused me a bit. I mean, why is it cloned if other objects aren't? 

So removing it a big +1 to me.